### PR TITLE
stop polluting args (e.g. row) with grid

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by albertywu** [comment]
:+1:

**Comment by bcherny** [comment]
I dont understand how this is different from the previous code - can you walk me through it?

**Comment by ghost** [comment]
I checked all consumers of the trigger function - they always pass an object for the args param.

File: src/index.js [hunk 1]
```diff
@@ -1341,8 +1341,7 @@ function SlickGrid(container, data, columns, options) {
 
   function trigger(evt, args, e) {
     e = e || new Slick.EventData();
-    args = args || {};
-    args.grid = self;
+    args = $.extend({}, args, {grid: self});
```
Comment (bcherny) [inline src/index.js:1344]: Talking it through in person, this ^ is just defensive.

File: src/index.js [hunk 2]
```diff
@@ -3135,7 +3134,7 @@ function SlickGrid(container, data, columns, options) {
 
     var activeRowChanged = (activeRow !== previousActiveRow);
     if (activeRowChanged) {
-      trigger(self.onActiveRowChanged, getDataItem(activeRow));
+      trigger(self.onActiveRowChanged, {row: getDataItem(activeRow)});
```
Comment (bcherny) [inline src/index.js:3137]: And this ^ is the actual fix.

**Review by bcherny — APPROVED** [review]

**Pull request merged**